network/netplan/netplan.go

```diff
@@ -135,15 +137,31 @@ func (np *Netplan) readYamlFile(path string) (err error) {
 	return nil
 }
 
-// ReadDirectory Reads the contents of a netplan directory and
+type sortableFileInfos []os.FileInfo
+
+func (fil sortableFileInfos) Len() int {
+	return len(fil)
+}
+
+func (fil sortableFileInfos) Less(i, j int) bool {
+	return fil[i].Name() < fil[j].Name()
+}
+
+func (fil sortableFileInfos) Swap(i, j int) {
+	fil[i], fil[j] = fil[j], fil[i]
+}
+
+// ReadDirectory reads the contents of a netplan directory and
 // returns complete config.
 func ReadDirectory(dirPath string) (np Netplan, err error) {
 	fileInfos, err := ioutil.ReadDir(dirPath)
 	if err != nil {
 		return np, err
 	}
 	np.sourceDirectory = dirPath
-	for _, fileInfo := range fileInfos {
+	sortedFileInfos := sortableFileInfos(fileInfos)
+	sort.Sort(sortedFileInfos)
+	for _, fileInfo := range sortedFileInfos {
```

Comment:
FWIW, I'm pretty sure you can just do:
sort.Sort(sortableFileInfos(fileInfos))
for _, fileInfo := range fileInfos {

because its just a type cast, it still operates on the underlying slice.